code/modules/projectiles/projectile/energy.dm

```diff
@@ -147,3 +147,63 @@
 	armour_penetration = 10 // It can have a little armor pen, as a treat. Bigger than it looks, energy armor is often low.
 	shield_buster = TRUE
 	is_reflectable = FALSE //I will let eswords block it like a normal projectile, but it's not getting reflected, and eshields will take the hit hard.
+
+/obj/item/projectile/energy/detective
+	name = "placeholder name"
+	icon_state = "omnilaser"
+	light_color = LIGHT_COLOR_CYAN
+	damage = 5
+	stamina = 25
+	eyeblur = 5
+
+/obj/item/projectile/energy/detective/overcharged
+	name = "overcharged shot"
+	icon_state = "spark"
+	light_color = LIGHT_COLOR_DARKRED
+	damage = 45
+	stamina = 15
+	eyeblur = 10
+
+/obj/item/projectile/energy/detective/tracker_warrant_shot
+	name = "tracker shot"
+	icon_state = "purple_laser"
+	light_color = LIGHT_COLOR_PURPLE
+	stamina = 0
+
+/obj/item/projectile/energy/detective/tracker_warrant_shot/on_hit(atom/target)
+	. = ..()
+	if(!ishuman(target))
+		no_worky(target)
+		return
+	start_tracking(target)
+	set_warrant(target)
+
+/obj/item/projectile/energy/detective/tracker_warrant_shot/proc/start_tracking(atom/target)
+	for(var/obj/item/gun/energy/detective/D in firer)
+		if(D.tracking_target_UID)
+			no_worky(tracking_already = TRUE)
+			return
+		D.start_pointing(target.UID())
+	qdel(src)
+
+/obj/item/projectile/energy/detective/tracker_warrant_shot/proc/set_warrant(atom/target)
+	var/mob/living/carbon/human/target_to_mark = target
+	var/perpname = target_to_mark.get_visible_name(TRUE)
+	if(!perpname || perpname == "Unknown")
+		no_worky(warrant_fail = TRUE)
+		return
+	var/datum/data/record/R = find_record("name", perpname, GLOB.data_core.security)
+	if(!R || R.fields["criminal"] == (SEC_RECORD_STATUS_EXECUTE || SEC_RECORD_STATUS_ARREST))
```

Comment:
this doesn't work
https://discord.com/channels/145533722026967040/727448459862343711/985614863571251270
```suggestion
	if(!R || R.fields["criminal"] in list(SEC_RECORD_STATUS_EXECUTE, SEC_RECORD_STATUS_ARREST))
```
this should though.